api/candlepin-api-spec.yaml

```diff
@@ -1776,6 +1895,17 @@ components:
               type: string
               example: /products/ff808081554a3e4101554a3e9033005d
 
+    ProvidedProductDTO:
+      description: ProvidedProduct DTO representation
+      required:
+        - productId
+      properties:
+        productId:
+          type: string
+          minLength: 1
```

Comment:
In order for the required: and minLength: restrictions to take effect, and to also validate against null ProvidedProductDTO elements in Sets, we should add these two calls: `DTOValidator.validateConstraints(dto)` and `DTOValidator.validateCollectionElementsNotNull(dto::getProvidedProducts, dto::getDerivedProvidedProducts)` to these two endpoints: `OwnerResource.createPool` and `OwnerResource.updatePool`.

Reply:
Done.